### PR TITLE
TextRender: add jsonpretty module

### PR DESCRIPTION
**Pull request by kwaegema**
add module to Indent json and add spaces before and after

### CODE REVIEW

File: src/main/perl/TextRender.pm [hunk 1]
```diff
@@ -422,17 +422,24 @@ sub load_module
...
+sub render_jsonpretty
+{
+    my ($self) = @_;
+    return $self->render_json(1);
+}
 
 sub render_json
 {
-    my ($self) = @_;
+    my ($self, $pretty) = @_;
```
Comment (stdweird) [inline src/main/perl/TextRender.pm]: add line of comment what this method and the optional arg does

File: src/main/perl/TextRender.pm [hunk 2]
```diff
@@ -422,17 +422,24 @@ sub load_module
...
 
     # We should only support hash or array refs (see JSON::XS allow_nonref option)
     # JSON::XS croaks if not handled properly
     my $ref = ref($self->{contents});
     if ($ref eq 'HASH' || $ref eq 'ARRAY') {
         my $j = JSON::XS->new();
         $j->canonical(1); # sort the keys, to create reproducable results
+        $j->pretty(1) if (defined($pretty) && $pretty);
```
Comment (stdweird) [inline src/main/perl/TextRender.pm]: just use `if !$pretty`, no need for `defined` as `undef` evaluates to false

Comment (stdweird) [reply src/main/perl/TextRender.pm]: it might even be better to do `$j->pretty($prettyr ? 1 : 0)` without `if`.

**Comment by kwaegema** [comment]
Remarks fixed

**Pull request merged**